vars/isUpstreamPRTrigger.groovy

```diff
@@ -0,0 +1,31 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+  Check it the build was triggered by an upstream job from a pull request.
+
+  def upstreamPRTrigger = isUpstreamPRTrigger()
+*/
+def call(){
+  def buildCause = currentBuild.getBuildCauses()?.find{ it._class == 'hudson.model.Cause$UpstreamCause'}
```

Comment:
I was tempted to call `isUpstreamTrigger` first, but then thought about the cost of iterating through all the build causes twice.

Reply:
I was thinking to change the signature of `isUpstreamTrigger` and support the filter something like

```
// isUpstreamTrigger
def call(Map args=[:]) {
  def filter = args.get('filter', 'all') 
  def buildCause = currentBuild.getBuildCauses()?.find{ it._class == 'hudson.model.Cause$UpstreamCause'}
  if (buildCause?.upstreamProject?.trim()) {
     // evaluate filter for PR
     if (filter!= 'all' ) {
       return buildCause?.upstreamProject?.toUpperCase()?.contains(filter))
     }
     return true
  }
  return false
}

```


the consumers

```

whenTrue(isUpstreamTrigger()) {
   //
}

whenTrue(isUpstreamTrigger(filter: 'PR-')) {
  //
}
```

Reply:
Do you mind if I add a commit rolling back this one, and adding that logic to the `isUpstreamTrigger` step?